File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    weight = [781, 156, 31, 6, 1]
    order = {'A':0, 'E':1, 'I':2, 'O':3, 'U':4}

    answer = 0
    for i in range(len(word)):
        answer += order[word[i]] * weight[i] + 1

    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {

        int[] weight = {781, 156, 31, 6, 1};
        int answer = 0;

        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);

            int order = 0;
            switch (c) {
                case 'A': order = 0; break;
                case 'E': order = 1; break;
                case 'I': order = 2; break;
                case 'O': order = 3; break;
                case 'U': order = 4; break;
            }

            answer += order * weight[i] + 1;
        }

        return answer;
    }
}